libraries/app/database_api.cpp

```diff
@@ -684,7 +685,7 @@ map<string,account_id_type> database_api_impl::lookup_accounts( const string& lo
                                                                 uint32_t limit,
                                                                 optional<bool> subscribe )const
 {
-   FC_ASSERT( limit <= 1000 );
+   FC_ASSERT( limit <= _app_options->api_limit_lookup_accounts );
```

Comment:
The error message or data need to include the value of the configured limit, otherwise the clients won't know what the limit is.

Sample response produced by current code:
```
$ curl -d '{"id":1,"method":"call","params":["database","lookup_accounts",["test",10000]]}' http://127.0.0.1:8090 2>/dev/null|
jq                                                                                                                                                                     
{
  "id": 1,
  "error": {
    "code": 10,
    "message": "Execution error",
    "data": {
      "code": 10,
      "name": "assert_exception",
      "message": "Assert Exception",
      "stack": [
        {
          "context": {
            "level": "error",
            "file": "database_api.cpp",
            "line": 688,
            "method": "lookup_accounts",
            "hostname": "",
            "thread_name": "th_a",
            "timestamp": "2020-01-29T12:07:48"
          },
          "format": "limit <= _app_options->api_limit_lookup_accounts: ",
          "data": {}
        }
      ]
    }
  }
}
```

All similar APIs need to be updated.

Reply:
Created issue https://github.com/bitshares/bitshares-core/issues/2141 to track this.